python/setup.py

```diff
@@ -0,0 +1,59 @@
+import os
+import subprocess
+import sys
+
+from setuptools import Extension, setup
+from setuptools.command.build_ext import build_ext
+
+# extend the Extension class to support CMake build
+
+class CMakeExtension(Extension):
+    def __init__(self, name):
+        super().__init__(name, sources=[])
+
+class CMakeBuildExtension(build_ext):
+    def build_extensions(self):
+        try:
+            out = subprocess.check_output(['cmake', '--version'])
+        except OSError:
+            raise RuntimeError('Cannot find CMake executable')
+
+        for ext in self.extensions:
+            self.build_cmake_ext(ext)
+        # super().run()
+
+    def build_cmake_ext(self, ext):
+
+        if not os.path.exists(self.build_temp):
+            os.makedirs(self.build_temp)
+
+        extdir = os.path.abspath(os.path.dirname(self.get_ext_fullpath(ext.name)))
+
+        # manually change LLVM_DIR and TiLT_DIR for the time-being
+        LLVM_DIR = "/home/jasperzhu/llvm-project-15.0.7/build/lib/cmake/llvm"
+        TILT_DIR = "/home/jasperzhu/tilt"
```

Comment:
lets not hardcode directory locations like this. make it parameterized. 
I don't think you need TILT_DIR location. You can get this from CMAKE_CURRENT_SOURCE_DIR env variable

Reply:
Also `setup.py` should be at the root directory. Keep only binding related source files inside `python/` directory.

Reply:
Moved setup.py to the root directory. LLVM_DIR is now specified using an environment variable.